doc/source/rllib/package_ref/trainer.rst

```diff
@@ -0,0 +1,50 @@
+.. _trainer-reference-docs:
+
+Trainer API
+===========
+
+The ``Trainer`` class is the highest-level API in RLlib. It allows you to
+train and evaluate policies, save an experiment's progress and restore from
+a prior saved experiment when continuing an RL run. ``Trainer`` is a sub-class
+of ``tune.Trainable`` and thus fully supports distributed hyperparameter tuning
+for RL.
+
+.. https://docs.google.com/drawings/d/1J0nfBMZ8cBff34e-nSPJZMM1jKOuUL11zFJm6CmWtJU/edit
+.. figure:: ../../images/rllib/trainer_class_overview.svg
+    :align: left
+
+    **A typical RLlib Trainer object:** The components sitting inside a Trainer are
+    normally: N ``WorkerSets`` (each consisting of one local ``RolloutWorker`` and zero or
+    more @ray.remote ``RolloutWorkers``),
```

Comment:
Can you link to these concepts? Or should you even introduce them here so early on?

I'm thinking that this seems a bit more like implementation. If a user is trying to understand how to use their Trainer, and they land on this page first, it will be quite confusing

Reply:
done

Reply:
I think introducing this here is very useful. There is often a confusion as to "Policy" vs "Trainer" vs "Worker", etc..

We do have the new svg here showing a few high-level components that go into a Trainer. I think this explains quite well, how RLlib handles distributed learning and what the "Policy"/"Trainer"/"Worker" concepts mean.

Reply:
Lined all classes mentioned to their code.